module/zfs/spa.c

```diff
@@ -2198,6 +2202,16 @@ spa_load_verify(spa_t *spa)
 		spa_load_note(spa, "spa_load_verify found %llu metadata errors "
 		    "and %llu data errors", (u_longlong_t)sle.sle_meta_count,
 		    (u_longlong_t)sle.sle_data_count);
+
+		// If the policy allows metadata errors,
+		// ignore the error returned by traverse_pool.
+		if (error == EINVAL &&
```

Comment:
`EINVAL` was the specific error I was getting from `traverse_pool` in the corrupted zpool I was testing with, so I thought it would be safer to apply the workaround only to that specific scenario, rather than to any `traverse_pool` error.  But if you're comfortable applying it more broadly, that's OK to me.  I changed it to `error != 0`.